gateway/api/server.go

```diff
@@ -705,4 +706,9 @@ func (api *Api) buildRoutes(r *apiroutes.Router) {
 		r.AuthMiddleware,
 		apiserverconfig.GenerateApiKey,
 	)
+
+	r.POST("/searches",
```

Comment:
I think `/searches` would make sense if we're treating this as a resource, like creating, listing and fetching a search.

I think it's more semantically appropriated to leave as `/search`.

Not a merge blocker, though.